pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2245,65 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	shouldFQLBeEnabled := false
+	dc := rc.GetDatacenter()
+	if dc.Spec.Config != nil {
+		var dcConfig map[string]interface{}
+		if err := json.Unmarshal(dc.Spec.Config, &dcConfig); err != nil {
+			rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+			return result.RequeueSoon(2)
+		}
+		casYaml, found := dcConfig["cassandra-yaml"]
+		if !found {
+			return result.Continue()
+		}
+		casYamlMap, ok := casYaml.(map[string]interface{})
+		if !ok {
+			err := fmt.Errorf("type casting error")
+			rc.ReqLogger.Error(err, "couldn't cast cassandra-yaml value from config to map[string]interface{}")
+			return result.RequeueSoon(2)
+		}
+		if _, found := casYamlMap["full_query_logging_options"]; found {
```

Comment:
`SetFullQueryLogging` is called after all C* pods should be ready. It would be nice to pull the validation part out into a separate function that is called earlier, preferably before creating/updating StatefulSets.

It would nice to have some unit tests as well.